README.md

```diff
@@ -82,6 +82,20 @@ See the [Reporter Interface section of the API documentation](https://github.com
 
 Looking for more examples? Check out the [examples folder](https://github.com/hapijs/good/tree/master/examples).
 
+## Reporter Type
```

Comment:
I find it contradicting to want on one hand to allow good to use any stream implementation and on the other not to check for common errors we currently have even in streams that were written explicitly to be used with good.

I came across two reporters completely failing because the maintainers did not understand  this issue. I did a quick search on npm and found another package that is probably affected as well.

I'm sure other maintainers and users will find this bugs eventually it just a bit hard to notice and will probably be only caught after code is in production (we were lucky enough to catch it on our staging env).

Maybe I'm completely missing the amount of effort it would take for adding such a check, but as good user I  would personally appreciate to get a heads up before using pipelines that are bound to fail when no one watching

Yet this is my personal opinion and if I'm the only one thinking it than it is probably indeed not worth the effort